app/components/primer/beta/link.rb

```diff
@@ -55,30 +52,29 @@ class Link < Primer::Component
       #       Link
       #     <% end %>
       #
-      # @param tag [String]  <%= one_of(Primer::Beta::Link::TAG_OPTIONS) %>
-      # @param href [String] URL to be used for the Link. Required if tag is `:a`. If the requirements are not met an error will be raised in non production environments. In production, an empty link element will be rendered.
+      # @param href [String] URL to be used for the Link. Required. If the requirements are not met an error will be raised in non production environments. In production, an empty link element will be rendered.
       # @param scheme [Symbol] <%= one_of(Primer::Beta::Link::SCHEME_MAPPINGS.keys) %>
       # @param muted [Boolean] Uses light gray for Link color, and blue on hover.
       # @param underline [Boolean] Whether or not to underline the link.
       # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-      def initialize(href: nil, tag: DEFAULT_TAG, scheme: DEFAULT_SCHEME, muted: false, underline: true, **system_arguments)
+      def initialize(href: nil, scheme: DEFAULT_SCHEME, muted: false, underline: true, **system_arguments)
         @system_arguments = system_arguments
 
         @id = @system_arguments[:id]
 
-        @system_arguments[:tag] = fetch_or_fallback(TAG_OPTIONS, tag, DEFAULT_TAG)
+        @system_arguments[:tag] = :a
```

Comment:
Can we use the [`deny_tag_argument`](https://github.com/primer/view_components/blob/9829b632e1f6c16262377fa2c7e5dcb7bc4d929b/app/components/primer/component.rb#L133) helper here to raise an error if a tag is passed? That should prevent us having to write a linter, which I think is a bit overkill.